File: core/src/org/rexellentgames/dungeon/debug/GiveCommand.java
package org.rexellentgames.dungeon.debug;

import org.rexellentgames.dungeon.UiLog;
import org.rexellentgames.dungeon.entity.creature.player.Player;
import org.rexellentgames.dungeon.entity.item.Item;
import org.rexellentgames.dungeon.entity.item.ItemHolder;
import org.rexellentgames.dungeon.entity.item.ItemRegistry;

import java.lang.reflect.Constructor;

public class GiveCommand extends ConsoleCommand {
	{
		shortName = "/gv";
		name = "/give";
		description = "[item] (count) gives an item";
	}

	@Override
	public void run(Console console, String[] args) {
		int count = 1;

		if (args.length == 2) {
			count = Integer.valueOf(args[1]);
		}

		if (args.length < 3) {
			String name = args[0];

			try {
				Class<?> clazz = ItemRegistry.items.get(name);

				if (clazz == null) {
					UiLog.instance.print("[red]Unknown item");
					return;
				}

				Constructor<?> constructor = clazz.getConstructor();
				Item item = (Item) constructor.newInstance(new Object[] { });
				ItemHolder itemHolder = new ItemHolder();
				itemHolder.setItem(item);

				if (item.isStackable()) {
					item.setCount(count);
				}

				Player.instance.getInventory().add(itemHolder);
			} catch (Exception e) {
				UiLog.instance.print("[red]Failed to create item, consult @egordorichev");
				e.printStackTrace();
			}
		} else {
			UiLog.instance.print("/give [item] (count)");
		}
	}
}

File: core/src/org/rexellentgames/dungeon/debug/GiveCommand.kt
package org.rexcellentgames.burningknight.debug

import org.rexcellentgames.burningknight.entity.creature.player.Player
import org.rexcellentgames.burningknight.entity.item.ItemHolder
import org.rexcellentgames.burningknight.entity.item.ItemRegistry
import org.rexcellentgames.burningknight.util.Log
import kotlin.reflect.full.createInstance

class GiveCommand : ConsoleCommand("/give", "/gv", "item] (count) gives an item") {
    override fun run(console: Console, args: Array<String>) {
        var count = 1

        if (args.size == 2) {
            count = Integer.valueOf(args[1])
        }

        if (args.size < 3) {
            val name = args[0]

            try {
                val clazz = ItemRegistry.items[name]
                if (clazz == null) {
                    Log.error("[Unknown item")
                    return
                }

                val item = clazz.createInstance()
                val itemHolder = ItemHolder()
                itemHolder.item = item

                if (item.isStackable) {
                    item.count = count
                }

                Player.instance.inventory.add(itemHolder)
            } catch (e: Exception) {
                Log.error("Failed to create item, consult @egordorichev")
                e.printStackTrace()
            }
        }
    }
}